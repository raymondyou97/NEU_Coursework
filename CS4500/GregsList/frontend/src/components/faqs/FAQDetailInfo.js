import React from 'react';
import PropTypes from 'prop-types';

const FAQDetailInfo = ({ faq, faqs, onChange }) => (
  <div>
    <div>
      <label>
        ID
        <select
          value={faq.id}
          onChange={e => onChange(e.target.value)}
          className="form-control"
        >
          {faqs.map(faq => (
            <option value={faq.id} key={faq.id}>
              {faq.id}
            </option>
          ))}
        </select>
      </label>
    </div>
    <div>
      <label>
        Title
        <input onChange={() => {}} className="form-control" value={faq.title} />
      </label>
    </div>
    <div>
      <label>
        Question
        <input
          onChange={() => {}}
          className="form-control"
          value={faq.question}
        />
      </label>
    </div>
  </div>
);

FAQDetailInfo.propTypes = {
  faqs: PropTypes.array,
  faq: PropTypes.shape({
    id: PropTypes.number.isRequired,
    title: PropTypes.string.isRequired,
    question: PropTypes.string.isRequired,
  }).isRequired,
  onChange: PropTypes.func.isRequired,
};

export default FAQDetailInfo;
