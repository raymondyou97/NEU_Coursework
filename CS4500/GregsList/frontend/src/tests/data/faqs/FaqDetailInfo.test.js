import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import FAQDetailInfo from '../../../components/faqs/FAQDetailInfo';
import { makeFaqs } from './FaqData';

test('FAQ Detail Components', () => {
  const faqs = makeFaqs();

  const component = shallow(
    <FAQDetailInfo faq={faqs[0]} faqs={faqs} onChange={() => {}} />
  );
  expect(component.find('option').length).toEqual(faqs.length);

  const titleValue = component.find('input');
  expect(titleValue.at(0).props().value).toEqual(faqs[0].title);
  expect(titleValue.at(1).props().value).toEqual(faqs[0].question);
});

test('Snapshot renders', () => {
  const faqs = makeFaqs();

  const component = renderer.create(
    <FAQDetailInfo faq={faqs[0]} faqs={faqs} onChange={() => {}} />
  );

  let componentTree = component.toJSON();
  expect(componentTree).toMatchSnapshot();
});
