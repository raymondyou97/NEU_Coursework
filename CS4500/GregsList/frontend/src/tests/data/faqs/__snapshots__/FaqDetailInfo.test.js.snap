// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot renders 1`] = `
<div>
  <div>
    <label>
      ID
      <select
        className="form-control"
        onChange={[Function]}
        value={0}
      >
        <option
          value={0}
        >
          0
        </option>
        <option
          value={1}
        >
          1
        </option>
        <option
          value={2}
        >
          2
        </option>
        <option
          value={3}
        >
          3
        </option>
        <option
          value={4}
        >
          4
        </option>
      </select>
    </label>
  </div>
  <div>
    <label>
      Title
      <input
        className="form-control"
        onChange={[Function]}
        value="Question #0"
      />
    </label>
  </div>
  <div>
    <label>
      Question
      <input
        className="form-control"
        onChange={[Function]}
        value="What is your favorite color?"
      />
    </label>
  </div>
</div>
`;
